lib/manager/composer/range.ts

```diff
@@ -42,7 +42,7 @@ export function getRangeStrategy(config: RangeConfig): RangeStrategy {
     logger.trace({ dependency: depName }, 'Pinning app require');
     return 'pin';
   }
-  if (isComplexRange) {
+  if (isComplexRange || ['typo3-cms-extension'].includes(composerJsonType)) {
```

Comment:
```suggestion
  if (isComplexRange || composerJsonType === 'typo3-cms-extension') {
```

Or do you expect there to be other types to be added in future?

Reply:
I would expect there to be more, yes. The argument for this change would probably extend to plugins of other PHP frameworks as well.